src/lib/proof_system.ts

```diff
@@ -467,6 +472,43 @@ function synthesizeMethodArguments(
   return args;
 }
 
+function methodArgumentsToConstant(
+  { allArgs, proofArgs, witnessArgs }: MethodInterface,
```

Comment:
two new helper functions we need to convert the input arguments of a zkapp call to inputs to our hash function